inc/uninstall.class.php

```diff
@@ -469,6 +473,46 @@ static function deleteFusionInventoryLink($itemtype, $items_id) {
       }
    }
 
+   /**
+    * Function to remove GLPI Inventory informations for an asset
+    *
+    * @param $itemtype the asset type
+    * @param $items_id the asset's ID in GLPI
+    *
+    * @return nothing
+   **/
+   static function deleteGlpiInventoryLink($itemtype, $items_id) {
+
+      $plug = new Plugin();
+      if ($plug->isActivated('glpiinventory') && function_exists('plugin_pre_item_purge_glpiinventory')) {
+         $item = new $itemtype();
+         $item->getFromDB($items_id);
+
+         // let glpi-inventory to clean item if needed (agent / collect etc ..)
+         plugin_pre_item_purge_glpiinventory($item);
+      } else {
+         self::purgeAgents($itemtype, $items_id);
```

Comment:
No need to split this into many methods.
```suggestion
         $agent = new Agent();
         $agent->deleteByCriteria([
            'itemtype' => $item->getType(),
            'items_id' => $item->getID(),
         ]);
```